src/test/java/com/zs/assignment9/StudentServiceTest.java

```diff
@@ -0,0 +1,80 @@
+package com.zs.assignment9;
+
+import com.zs.assignment9.dao.StudentDAOImpl;
+import com.zs.assignment9.exception.BadRequestException;
+import com.zs.assignment9.model.Student;
+import com.zs.assignment9.service.StudentServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.stream.Stream;
+
+
+@ExtendWith(MockitoExtension.class)
+class StudentServiceTest {
+
+    @Mock
+    private StudentDAOImpl studentDAO;
+
+    @InjectMocks
+    private StudentServiceImpl studentService;
+
+
+    static Stream<Arguments> getData() {
+        return Stream.of(
+                Arguments.arguments(1,"sam","ram",false),
+                Arguments.arguments(-1,"sam","ram", true),
+                Arguments.arguments(1,null,"sam",true),
+                Arguments.arguments(1,"sam",null,true)
+        );
+    }
+
+
+    @Test
+    void getStudent() throws BadRequestException {
```

Comment:
don't throw exceptions from test methods catch them and do fail with exceptions message